context/QuestsProvider.tsx

```diff
@@ -57,38 +57,40 @@ export const QuestsContextProvider = ({
 
   useMemo(() => {
     (async () => {
-      const data: GetQuestsRes = await getQuests();
-
-      const q = Object.values(data).flat();
-
-      const categoriesWithImages = await Promise.all(
-        Object.keys(data).map(async (key) => {
-          const img = await (async () => {
-            try {
-              // If a category img is defined in quest_categories use it
-              const questData = await fetchQuestCategoryData(key);
-              return questData.img_url;
-            } catch (error) {
-              // else use img from first quest in the category
-              return q.filter((quest) => quest.category === key)[0].img_card;
-            }
-          })();
-          const questNumber = q.filter(
-            (quest) => quest.category === key
-          ).length;
-          const quests = q.filter((quest) => quest.category === key);
-
-          return {
-            name: key,
-            img,
-            questNumber,
-            quests,
-          };
-        })
-      );
-
-      setCategories(categoriesWithImages);
-      setQuests(q);
+      const data: GetQuestsRes | undefined = await getQuests();
+
+      if (data) {
```

Comment:
`if (!data || "error" in res) return;`

// add code of if block below

can work here when you add QueryError instead of undefined